### DISCUSSION TITLE
WorkflowName and triggering event on checkRun API

### DISCUSSION THREAD

**Discussion by qfc** [community/community#26191] [Actions]
<p>There is no additional information about which worklfow a checkRun belongs to on checkRunApi</p>
<p>And there is no information about which event triggered the checkRun. Events like pull_request/push</p>
<p>Any help would be appreciated. </p>
<p><a href="https://developer.github.com/v3/checks/runs/#get-a-single-check-run" rel="nofollow noopener">https://developer.github.com/v3/checks/runs/#get-a-single-check-run</a></p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/qfc">@qfc</a> ,</p>
<p>checkrun rest api doesn’t contain workflow name aknd triggering event type.  These information can be found in <a href="https://help.github.com/en/actions/reference/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">github context</a> content. You can get the workflow name and event type with below code:</p>
<pre><code>steps:
      - name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"
      - name: check workflowname
        run: |
          echo ${{ github.workflow }}
      - name: check eventname
        run: |
          echo ${{ github.event_name }}
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181096003-80ad1f04-b312-4d24-b52f-8126cfe44b4d.png" alt="workflownameandevent.png" data-base62-sha1="qwRkNGt3sA2nTaSknFm6IAxiRJQ" width="469" height="341"></p>
<p>If you’d like any improvement of checkrun rest api, it’s recommended to raise your idea <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where github manager will take a review.</p>
<p>Thanks!</p>

**Comment by qfc** [comment]
<p>Thanks a lot!</p>
